tests/sqlite/sqlite.rs

```diff
@@ -725,3 +725,22 @@ async fn concurrent_read_and_write() {
     read.await;
     write.await;
 }
+
+#[sqlx_macros::test]
+async fn test_query_with_progress_handler() -> anyhow::Result<()> {
+    let mut conn = new::<Sqlite>().await?;
+    conn.lock_handle()
+        .await?
+        .set_progress_handler(1, || false)
+        .await;
```

Comment:
**Please** include some sort of captured state in this closure so it tests correct allocation/deallocation of the callback, e.g.:

```suggestion
    // Using this string as a canary to ensure the callback doesn't get called with the wrong data pointer.
    let string = format!("Hello from the progress handler!");

    conn.lock_handle()
        .await?
        .set_progress_handler(1, move || { 
            println!("{}", state);
            false 
        });
```